### DISCUSSION TITLE
Replace nvpmodel.conf with a custom one in dunfell-l4t-r32.6.1

### DISCUSSION THREAD

**Discussion by notAstereo** [OE4T/meta-tegra#1252] [Q&A] [answered]
Hi

I want to replace the default `nvpmodel.conf` file. That has been working fine using the trick described in https://github.com/OE4T/meta-tegra/issues/89 where you append the recipes for `tegra-binaries` and `tegra-nvpmodel-base`. That was on `dunfell32.4.3`.

However, now I've switched to `dunfell32.6.1` and this trick it not working anymore. The error I'm getting is 
```
| install: cannot stat '/home/myhomedir/repos/meta-mymetalayer/build/tmp/work/jetson_tx2_devkit-poky-linux/tegra-nvpmodel-base/32.6.1-20210726122859-r0/tegra-nvpmodel-base-32.6.1-20210726122859/config-replacements/nvpmodel.conf': No such file or directory
```
Apparently, the file `nvpmodel.conf` cannot be found. I am also surprised how this trick even worked in the first place as I thought that recipes don't interact with each other.
I am building for a `Jetson TX2`.

Is there another recommended way to get a custom `nvpmode.conf` into the build? 
Can this trick be fixed?

**Comment by madisongh** [comment, marked as answer]
Starting with R32.6.1, we switched to populating most of the BSP files using NVIDIA's deb packages, rather than extracting them from the L4T tarball.  That should simplify things, as you can use a `bbappend` directly on the `tegra-nvpmodel` recipe to add your custom file to the SRC_URI, like you would do with most other recipes.

**Comment by notAstereo** [comment]
Thank you! That helped me.
I ended up appending to the `tegra-nvpmodel-base` since this is the one that deals with the nvpmodel conf file. Using the `tegra-nvpmodel` directly caused a error when building the image. Something along the lines: "There are two recipes trying to install the same file".
I also opted to append to the nvpmodel config instead of replacing it completely.

In `tegra-nvpmodel-base_%.bbappend`
do

```
FILESEXTRAPATHS_prepend := "${THISDIR}/files:"

SRC_URI += "file://nvpmodel.txt"

NVPMODEL_CONFIG_DEFAULT= "5"
do_install:append:tegra186() {
        cat ${WORKDIR}/nvpmodel.txt >> ${D}${sysconfdir}/nvpmodel.conf
}
```

**Reply by madisongh** [reply]
Oops, right, that's the one.  I had to split some of the recipes so we could add stuff from the layer on top of what's in the L4T packages.
